main.py

```
@@ -1,29 +1,106 @@
+                "Invalid path to directory : ", weather_data_folder_path
+            )
+
+        report_generator = ReportGenerator()
+
+        if args.e:
+            data_provider = ReportDataProvider(args.e, weather_data_folder_path)
+            report_data = data_provider.get_high_low_temp_high_humidity_data()
+
+            report_generator.gen_report_highest_lowest_temp_and_humidity(
+                report_data
+            )
+
+        if args.a:
+            data_provider = ReportDataProvider(args.a, weather_data_folder_path)
+            report_data = data_provider.get_average_temp_and_humidity()
 
-    weather_data_files_path = CURRENT_DIRECTORY + sys.argv[1]
-    if not os.path.isdir(weather_data_files_path):
-        print("Invalid path to directory : ", weather_data_files_path)
-        sys.exit(0)
+            report_generator.gen_report_average_max_min_temp_and_mean_humidity(
+                report_data
+            )
 
-    print(weather_data_files_path)
+        if args.c:
+            data_provider = ReportDataProvider(args.c, weather_data_folder_path)
+            report_data = data_provider.get_chart_data()
 
-    system_args = sys.argv[2:]
+            report_generator.gen_report_high_low_temperature_charts(
+                report_data
+            )
 
-    year = sys.argv[3]
+        if args.cs:
+            data_provider = ReportDataProvider(
+                args.cs, weather_data_folder_path
+            )
+            report_data = data_provider.get_chart_data()
 
-    file_path_matcher = FilePathMatcherWithDate(weather_data_files_path)
-    file_path_matcher.setDate(year)
-    matched_file_paths_with_year = file_path_matcher.get_files_path()
+            report_generator.gen_report_high_low_temperature_single_line_chart(
+                report_data
+            )
 
-    # for index, argument in enumerate(system_args):
-    #     pass
+    except Exception as e:
+        print(e)
+        print(traceback.format_exc())
```

raise the exception instead of printing it